Libraries/Image/RCTImageLoader.m

```diff
@@ -425,17 +425,19 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
 
       dispatch_async(_URLCacheQueue, ^{
 
-        // Cache the response
-        // TODO: move URL cache out of RCTImageLoader into its own module
-        BOOL isHTTPRequest = [request.URL.scheme hasPrefix:@"http"];
-        [strongSelf->_URLCache storeCachedResponse:
-         [[NSCachedURLResponse alloc] initWithResponse:response
-                                                  data:data
-                                              userInfo:nil
-                                         storagePolicy:isHTTPRequest ? NSURLCacheStorageAllowed: NSURLCacheStorageAllowedInMemoryOnly]
-                                        forRequest:request];
-        // Process image data
-        processResponse(response, data, nil);
+        if(response) {
+          // Cache the response
+          // TODO: move URL cache out of RCTImageLoader into its own module
+          BOOL isHTTPRequest = [request.URL.scheme hasPrefix:@"http"];
+          [strongSelf->_URLCache storeCachedResponse:
+           [[NSCachedURLResponse alloc] initWithResponse:response
+                                                    data:data
+                                                userInfo:nil
+                                           storagePolicy:isHTTPRequest ? NSURLCacheStorageAllowed: NSURLCacheStorageAllowedInMemoryOnly]
+                                          forRequest:request];
+          // Process image data
+          processResponse(response, data, nil);
```

Comment:
You are right. It's better to call completion handler and return when the "response" is nil.
I will change the code, as well as the "if format" you noted above.